presto-spi/src/main/java/com/facebook/presto/spi/type/Decimals.java

```diff
@@ -176,7 +178,7 @@ public static String toString(long unscaledValue, int scale)
 
     public static String toString(Slice unscaledValue, int scale)
     {
-        return toString(decodeUnscaledValue(unscaledValue), scale);
+        return toString(toStringDestructive(unscaledDecimal(unscaledValue)), scale);
```

Comment:
Instead of exposing `toString*Destructive*`, make the public method non-destructive and have it create the copy internally.